client/client.go

```diff
@@ -2294,13 +2296,22 @@ func (client *OpenFgaClient) singleBatchCheck(ctx _context.Context, body fgaSdk.
 	}
 
 	requestID := fmt.Sprintf("%d", time.Now().UnixNano())
+	headerName := "X-OpenFGA-Client-Bulk-Request-Id"
 
-	client.APIClient.GetConfig().AddDefaultHeader("X-OpenFGA-Client-Bulk-Request-Id", requestID)
+	// Create a mutex-protected copy of headers to avoid race conditions
```

Comment:
is the possible race condition that another caller attempts to mute the config by adding a default header at the same time? Was this observed during testing or just identified by inspection?

Reply:
The race condition popped up after adding the BatchCheckMaxBatchSize test with the tests in the go-sdk repo.

Reply:
Resolving as we've reverted the header setting